waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -324,6 +325,48 @@ validateCrudOperations spec =
         maybeIdBlockAttribute = Entity.getIdBlockAttribute entity
         (entityName, entity) = AS.resolveRef spec (AS.Crud.entity crud)
 
+validatePrismaOptions :: AppSpec -> [ValidationError]
+validatePrismaOptions spec =
+  concat
+    [ checkIfPostgresExtensionsAreUsedWithoutPostgresDbSystem,
+      checkIfDbExtensionsAreUsedWithoutPostgresDbSystem,
+      checkIfDbExtensionsAreUsedWithoutPostgresExtensions
+    ]
+  where
+    prismaOptions = AS.Db.prisma =<< AS.App.db (snd $ getApp spec)
+
+    isPostgresDbUsed = isPostgresUsed spec
+
+    checkIfPostgresExtensionsAreUsedWithoutPostgresDbSystem :: [ValidationError]
+    checkIfPostgresExtensionsAreUsedWithoutPostgresDbSystem = case prismaClientPreviewFeatures of
+      Nothing -> []
+      Just value ->
+        [ GenericValidationError "You enabled \"postgresqlExtensions\" in app.db.prisma.clientPreviewFeatures but your db system is not PostgreSQL."
+          | not isPostgresDbUsed && "postgresqlExtensions" `elem` value
+        ]
+
+    checkIfDbExtensionsAreUsedWithoutPostgresDbSystem :: [ValidationError]
+    checkIfDbExtensionsAreUsedWithoutPostgresDbSystem = case prismaDbExtensions of
+      Nothing -> []
+      Just value ->
+        [ GenericValidationError "You are using app.db.prisma.dbExtensions but your db system is not PostgreSQL."
```

Comment:
This doesn't really it make it clera why is that wrong. Maybe you could mention that PostgreSQL is required in order to specify extensions, currently.